### PR TITLE
Heating water coil sizing mixing up hot and cold initialization temperatures

### PR DESCRIPTION
**Pull request by Nigusse**
Issue #5821 and #6502

Pull request overview
---------------------
This fix replaces water initialization temperature for Hot Water related sizing calculations in various modules. Historically DataGlobals::CWInitConvTemp (5.05C) value was used in several modules for chilled and hot water related sizing calculations for flow rate, capacity, coil UA, etc. This fix now replaces DataGlobals::CWInitConvTemp with DataGlobals::HWInitConvTemp(60.0C) for hot water related sizing calculations. Changes were made in several functions including:  InitGasAbsorber, SizeAbsorpChiller, InitBLASTAbsorberModel, InitExhaustAbsorber, InitIndirectAbsorpChiller, SizeIndirectAbsorpChiller, InitElecReformEIRChiller, InitDesiccantDehumidifier, InitFurnace, InitGshp, HVACFourPipeBeam, GetMSHeatPumpInput, InitCBVAV, InitUnitarySystems, InitHWBaseboard, PackagedTerminalHeatPump::InitPTUnit, PLantChillers::InitElectricChiller, SingleDuct::InitSys. The changes made include in sizing calculations in hot water coils, hot water baseboard and radiators, chillers heat recovery, absorption chillers generators. So, hot water related sizing and output variables are expected to show diffs. Also several unit tests failed due to this fix were also addressed.  

Also replaced hard value 5.0C with DataGlobals::CWInitConvTemp (5.05C) in LowTempRadiantSystem, ReportSizingManager, OutdoorAirUnit, and WaterCoils.Unit.cc modules. This change produces small diffs. Selective review of various EIO and output files shown expected diffs. 

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by Nigusse** [comment]
Sample Sizing results (EIO file) for 5ZoneAirCooled.idf example file below. The maximum diffs in the rated water coil heating capacity is 1.67%. In all cases the water coil rated capacity decreased primarily due to a decrease in hot water specific heat value with an increased water temperature as expected. On the other hand the design hot water volume flow rates increased due to a decrease in water density and a decrease in specific heat with increased sizing water temperature used. The sizing results diffs observed are expected trend consistent with an increase water temperature used for sizing calculation.

5821_5zoneaircooled_eio_diffs

**Comment by Nigusse** [comment]
Sample sizing calc results (EIO file) for 5zonecav_maxtemp.idf example file below. The hot water vol flow rate increased by about 1.3%-1.7% which is consistent with decrease in water density due to an increase of water temperature from 5.05C to 60.0C. The capacity in all cases slightly reduced due to decrease in specific heat of water as the temperature of water increased. These sizing results diffs observed are expected trend.

5821_5zonecav_maxtemp_eio_diffs

**Comment by Nigusse** [comment]
The unit test OutdoorAirUnit_AutoSize is shown as failed by the regression test but it passes in my Win64 machine run. There is no way for me to diagnose why it fails on Linux-Ubuntu-14.04 machine. @Myoldmopar Can you please look why this test failing when you get a chance?

**Comment by Nigusse** [comment]
Sample sizing calc results (EIO file) for WindACAuto.idf example file below. The Hot Water Baseboard convector hot water vol flow rate increased by about 1.7% which is consistent with decrease in water density and due to increase of water temperature from 5.05C to 60.0C. These sizing results diffs observed are expected trend.


5821_windacauto_eio_diffs

**Comment by Nigusse** [comment]
@mjwitte @Myoldmopar There are a few places in the code that use 5.0C for sizing calculation instead of CWInitConvTemp (5.05C). Do you have any objection if I replaces the 5.0C values with CWInitConvTemp?

**Comment by Myoldmopar** [comment]
@Nigusse I just merged in #6811 that temporarily disables a couple problematic unit tests.  I'm going to be working on getting them re-enabled and fixed up properly, but if you pull develop into this branch, that failure should be gone now.

**Comment by Nigusse** [comment]
@Myoldmopar Thanks. I just pulled develop into this branch.

**Comment by mjwitte** [comment]
@Nigusse Yes, please replace the hard 5.0 values.

**Comment by Nigusse** [comment]
@mjwitte @Myoldmopar Hot Water Boiler uses boiler hot water outlet temperature to calculate CP for nominal capacity sizing [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/Boilers.cc#L820-L828)

Boiler(BoilerNum).TempDesBoilerOut will be replaced with HWInitConvTemp for consistency.

**Comment by mjwitte** [comment]
@Nigusse Well, if you do that, then you should delete an input field.  See #4431

**Comment by Nigusse** [comment]
@mjwitte OK. I will replace Boiler(BoilerNum).TempDesBoilerOut in this fix but will deprecate the field in issue #4431.

**Comment by Nigusse** [comment]
@Myoldmopar @mjwitte This branch is ready for review and merge.

File: src/EnergyPlus/OutdoorAirUnit.cc [hunk 1]
```diff
@@ -1299,7 +1299,7 @@ namespace OutdoorAirUnit {
                         OutAirUnit(OAUnitNum).OAEquip(compLoop).MaxVolWaterFlow = GetWaterCoilMaxFlowRate(
                             OutAirUnit(OAUnitNum).OAEquip(compLoop).ComponentType, OutAirUnit(OAUnitNum).OAEquip(compLoop).ComponentName, errFlag);
                         rho = GetDensityGlycol(PlantLoop(OutAirUnit(OAUnitNum).OAEquip(compLoop).LoopNum).FluidName,
-                                               60.0,
+                                               DataGlobals::CWInitConvTemp,
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutdoorAirUnit.cc:1302]: Hooray for the removed magic numbers!

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 2]
```diff
@@ -1127,7 +1127,7 @@ namespace HWBaseboardRadiator {
                 } else if (HWBaseboard(BaseboardNum).RatedCapacity == AutoSize || HWBaseboard(BaseboardNum).RatedCapacity == 0.0) {
                     DesCoilLoad = RatedCapacityDes;
                     rho = GetDensityGlycol(PlantLoop(HWBaseboard(BaseboardNum).LoopNum).FluidName,
-                                           CWInitConvTemp,
+                                           DataGlobals::HWInitConvTemp,
```
Comment (Myoldmopar) [inline src/EnergyPlus/HWBaseboardRadiator.cc:1130]: Hooray for the scoped usage!

File: tst/EnergyPlus/unit/BoilerHotWater.unit.cc [hunk 3]
```diff
@@ -100,7 +100,7 @@ TEST_F(EnergyPlusFixture, Boiler_HotWaterSizingTest)
     // see if boiler volume flow rate returned is autosized value
     EXPECT_NEAR(Boilers::Boiler(1).VolFlowRate, 1.2, 0.000001);
     // see if boiler nominal capacity returned is autosized value
-    EXPECT_NEAR(Boilers::Boiler(1).NomCap, 50409257.0, 1.0);
+    EXPECT_NEAR(Boilers::Boiler(1).NomCap, 49376304.0, 1.0);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/BoilerHotWater.unit.cc:103]: Makes good sense.

File: tst/EnergyPlus/unit/HVACFourPipeBeam.unit.cc [hunk 4]
```diff
@@ -1689,8 +1695,8 @@ TEST_F(EnergyPlusFixture, Beam_sizeandSimulateOneZone)
 
     EXPECT_DOUBLE_EQ(DataLoopNode::Node(15).Temp, 14.0);
     EXPECT_DOUBLE_EQ(DataLoopNode::Node(15).MassFlowRate, 0.0);
-    EXPECT_NEAR(DataLoopNode::Node(39).Temp, 35.064466069323743, 0.00001);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/HVACFourPipeBeam.unit.cc:1692]: These are definitely the types of changes I was expecting.  It's awesome to see the unit tests in action like this.

**Review by Myoldmopar — APPROVED** [review]
Looks great!

**Comment by Myoldmopar** [comment]
Dug through the diffs a bit.  For example, in the 2ZoneDataCenter file, the absolute diffs show a highest change in power of 0.57 W.  This is definitely over the diff threshold, so the regression scripts are right to warn us, but these are the types of diffs I was expecting from an initialization temperature change.  This looks good to go.  I am re-running the full set of unit tests locally with develop pulled in just to make sure, but this should be good to go.

**Comment by Myoldmopar** [comment]
Yep, tests pass with develop merged in, this is done.

**Pull request merged**